server/src/database/models/course.ts

```diff
@@ -8,8 +8,6 @@ import { sequelize } from '..';
 export default class Course extends Model<InferAttributes<Course>, InferCreationAttributes<Course>> {
   declare id: CreationOptional<number>;
   declare courseCode: string;
```

Comment:
Not concerning the changes you made but should we enforce `unique` in the `courseCode`? While testing I could create multiple courses with same `courseCode`. This could create issues once we start adding course instances. To which course should we assign the instances if there are multiple course entries with the same `courseCode`? 

If we query instances based on the courses `id` it would only return instances signed to that `id`. If we query with `courseCode`, then it would return all instances no matter what the `id` is.

Reply:
This goes back to the question of whether course codes are unique at Aalto to begin with. I recall not getting a definitive answer. The immediate reaction was to assume that they are unique, but there might have been some cases where they are not with courses that have multiple language versions of them.

We could triple check this again, but I think sticking to using our internal numeric course ID for all logic is the safest option in any case if something changes in the future.

Reply:
If I remember correct, they said that we can not assume that they are unique, because there may be exceptions to the rule. That's why we finally ended up using a separate, auto-incrementing numeric id for courses.

Reply:
> The immediate reaction was to assume that they are unique, but there might have been some cases where they are not with courses that have multiple language versions of them.

Shouldn't the different language versions be defined in instance level, not in course level?

But anyway, I think we should triple check this thing with PO.

Reply:
But as mentioned initially too, this discussion doesn't particularly concern the PR, so it shouldn't block it from being merged and I think can be marked as resolved.